src/pages/index.js

```diff
@@ -58,16 +95,88 @@ addNewCardButton.addEventListener("click", () => {
 });
 
 function handleNewCardSubmit(name, link) {
-  renderCard({ name, link });
+  newCardPopup.handleLoading(true);
+  api
+    .createCard({ name, link })
+    .then((res) => {
+      cardSection.addItem(res);
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err); // log the error to the console
+    })
+    .finally(() => {
+      setTimeout(() => {
+        newCardPopup.handleLoading(false);
+      }, 2500);
```

Comment:
Needs correction: Please remove or reduce the timeout, you don't need it. The button text remains `Loading` too long after the submission

Reply:
Timeout has been reduced.